common/network_manager_test.go

```diff
@@ -123,3 +142,69 @@ func TestOnRequestPaused(t *testing.T) {
 		})
 	}
 }
+
+func TestOnRequestPausedBlockedIPs(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name, reqURL            string
+		blockedIPs, expCDPCalls []string
+	}{
+		{
+			name:        "ok_fail_simple",
+			blockedIPs:  []string{"10.0.0.0/8", "192.168.0.0/16"},
+			reqURL:      "http://10.0.0.1:8000/",
+			expCDPCalls: []string{"Fetch.failRequest"},
+		},
+		{
+			name:        "ok_fail_resolved_ip",
+			blockedIPs:  []string{"127.0.0.10/32"},
+			reqURL:      fmt.Sprintf("http://%s/", mockHostname),
+			expCDPCalls: []string{"Fetch.failRequest"},
+		},
+		{
+			name:        "ok_continue_resolved_ip",
+			blockedIPs:  []string{"127.0.0.50/32"},
+			reqURL:      fmt.Sprintf("http://%s/", mockHostname),
+			expCDPCalls: []string{"Fetch.continueRequest"},
+		},
+		{
+			name:        "ok_continue_simple",
+			blockedIPs:  []string{"127.0.0.0/8"},
+			reqURL:      "http://10.0.0.1:8000/",
+			expCDPCalls: []string{"Fetch.continueRequest"},
+		},
+		{
+			name:        "ok_continue_empty",
+			blockedIPs:  nil,
+			reqURL:      "http://127.0.0.1/",
+			expCDPCalls: []string{"Fetch.continueRequest"},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			blockedIPs := make([]*k6lib.IPNet, 0, len(tc.blockedIPs))
+			for _, ipcidr := range tc.blockedIPs {
+				ipnet, err := k6lib.ParseCIDR(ipcidr)
+				require.NoError(t, err)
+				blockedIPs = append(blockedIPs, ipnet)
+			}
```

Comment:
?

```suggestion
			blockedIPs := make([]*k6lib.IPNet, len(tc.blockedIPs))
			for i, ipcidr := range tc.blockedIPs {
				ipnet, err := k6lib.ParseCIDR(ipcidr)
				require.NoError(t, err)
				blockedIPs[i] = ipnet
			}
```

Reply:
Sure, but is there a practical benefit to this, or is it just personal preference?

Reply:
1. Idiomatic (check Go stdlib)
2. Simpler
3. More natural
4. Your code doesn't need to `append` as the size is there, you just want to fill its values
5. You don't need a function call which is designed for appending (and adds unnecessary assembly instructions)
6. ...